packages/repository/src/repositories/relation.factory.ts

```diff
@@ -4,21 +4,29 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {EntityCrudRepository} from './repository';
-import {RelationType} from '../decorators/relation.decorator';
+import {RelationType, RELATIONS_KEY} from '../decorators/relation.decorator';
 import {Entity} from '../model';
 import {
   HasManyEntityCrudRepository,
   DefaultHasManyEntityCrudRepository,
 } from './relation.repository';
+import {DefaultCrudRepository} from './legacy-juggler-bridge';
+import {MetadataInspector} from '@loopback/context';
 
 export interface RelationDefinitionBase {
   type: RelationType;
-  keyTo: string;
 }
 
 export interface HasManyDefinition extends RelationDefinitionBase {
   type: RelationType.hasMany;
+  keyTo: string;
+  keyFrom: string;
 }
+
+export type constrainRepositoryFunction<T extends Entity> = (
```

Comment:
Please rename this type to communicate the fact that this function is for HasMany relation only.

Suggested name:

`HasManyRepositoryFactory`